ansible/roles/satellite-manage-manifest/tasks/version_6.x.yml

```diff
@@ -0,0 +1,36 @@
+---
+- name: Check if manifest file is uploaded
+  stat:
+    path: "/tmp/{{ manifest_file | basename }}"
+  register: manifest_file_existance_result
+
+- name: Copy manifest
+  copy:
+    src: "{{ manifest_file }}"
+    dest: /tmp
+  when:
+    - not manifest_file_existance_result.stat.exists
+    - "not manifest_file.startswith('http')"
+    - "not manifest_file.startswith('ftp')"
+
+- name:
+  get_url:
+    url: "{{ manifest_file }}"
+    dest: /tmp
+  when:
+    - not manifest_file_existance_result.stat.exists
+    - "manifest_file.startswith('http') or manifest_file.startswith('ftp')"
```

Comment:
`manifest_file`  suggests it's a filename or file content.

I recommend you add another variable, `manifest_url` or something.
Using the same for both file and url is errorprone.

You can decide what method to use depending on the existence of the variables, or simply add a `manifest_method`.

You can also use a dictionary

```yaml
manifest_info:
  type: file
  content: "{{ ... }}"

manifest_info:
  type: url
  url: http://...
```

And do things accordingly, depending of the value of `manifest_info.type`

Reply:
Yeah, that's a nice idea :+1: